components/contributors/InfoCard.tsx

```diff
@@ -5,16 +5,28 @@ import Image from "next/image";
 import Link from "next/link";
 import { BsSlack } from "react-icons/bs";
 import { env } from "@/env.mjs";
+import { getLeaderboardData } from "@/app/api/leaderboard/functions";
+import { parseDateRangeSearchParam } from "@/lib/utils";
 
-export default function InfoCard({
+export default async function InfoCard({
   contributor,
-  minimal = true,
+  showRank,
   isClickable = false,
 }: {
   contributor: Contributor;
-  minimal?: boolean;
+  showRank?: boolean;
   isClickable?: boolean;
 }) {
+  const leaderboardData = await getLeaderboardData(
+    parseDateRangeSearchParam(),
+    "points",
+    "desc",
+    [],
+  );
+  const userPosition = leaderboardData.findIndex(
+    (data) => data.user.social.github === contributor.github,
+  );
```

Comment:
This compute is performed for each and every render of `InfoCard` component. For example, in the home page, we are showing this card N number of times, but we are not using the rank computed from this.

This could make it slow when we have larger amount of data.

Instead move the compute to the parent component and pass the rank down to this component only in places where it is required.

Reply:
If we fetch in the `contributor's` page which is the parent of `Infocard`. On that page, we have the `getContributorBySlug()` function and In the same page if we fetch from leaderboard data using `getLeaderboardData()` it creates partial deadlock. Right?

So, To resolve the performance issue we can apply conditional fetching if the show rank is true only then fetch data from the leaderboard. We can do this using the `useEffect` or `usePathname` as well.

Reply:
The deadlock wouldn't happen in this case since both are independently executed.